design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+type Query {
+
+    getNegotiableQuote(
+        id: ID!
+    ): NegotiableQuote @doc(description:"Get negotibale quote")
+
+    getAllNegotiableQuotes: [NegotiableQuote]! @doc(description: "Get all negotiable quotes")
+}
+
+type Mutation {
+
+    updatedPrices(
+        updatedQuotes: UpdatedPricesInput!
+    ): UpdatedPricesOutput! @doc(description:" This call refreshes item prices, taxes, discounts, cart rules in the negotiable quote. Quotes that are locked for the seller will not be updated.")
+
+    requestQuote(
+        negotiableQuote: NegotiableQuoteRequestInput!
+    ): RequestQuoteOutput! @doc(description:"Generates negotiable quote request")
+
+    assignBillingAddressToNegotiableQuote(
+        billingAddress: NegotiableQuoteBillingAddressInput!
+    ): AssignBillingAddressToNegotiableQuoteOutput @doc(description:"Assigns new billing address to negotiable quote")
+
+    estimateShippingCostForNegotiableQuote(
+        shippingAddress: NegotiableQuoteEstimateShippinAddressInput!
+    ): EstimateShippingCostForNegotiableQuoteOutput @doc(description:"Estimate shipping costs for a negotiable quote")
+
+    estimateShippingCostByAddressId(
+        shippingAddressId: NegotiableQuoteEstimateShippinAddressIdInput!
+    ): EstimateShippingCostByAddressIdOutput @doc(description:"Estimate shipping costs for a negotiable quote by Address ID")
+
+    estimateShippingInformationForNegotiableQuote(
+        addressInformation: NegotiableQuoteShippingInformationInput!
+    ): EstimateShippingInformationForNegotiableQuoteOutput @doc(description:"Get list of payment options and the order totals by address and shipping info")
+
+    setShippingMethodToNegotiableQuote(
+        negotiableQuoteShippingMethod: NegotiableQuoteShippingMethodInput!
+    ): setShippingMethodToNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+
+    setCouponCodeToNegotiableQuote(
+        negotiableQuoteCouponCode: NegotiableQuoteCouponCodeInput!
+    ): SetCouponCodeToNegotiableQuoteOutput @doc(description:"Set negotiable quote coupon code")
+
+    removeCouponCodeFromNegotiableQuote(
+        cartId: String!
+    ): RemoveCouponCodeFromNegotiableQuoteOutput @doc(description:"Remove negotiable quote coupon code")
+
+    setPaymentInformationForNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): SetPaymentInformationForNegotiableQuoteOutput! @doc(description:"Sets payment information and the billing address for the negotiable quote")
+
+    createOrderWithPaymentInfoNegotiableQuote(
```

Comment:
@mbrinton01 please confirm that it should not be possible to create an order from the negotiable quote directly, instead it should first be converted to cart and  then the user should follow the checkout process.

Reply:
This question is related to `createOrderWithGiftCardForNegotiableQuote` and `removeGiftCardFromNegotiableQuote` as well

Reply:
@paliarush I'm not sure of the exact method used now, but we should follow the same approach. It should not be any different for GraphQL.

Reply:
Yes, Nqgotiable Quote should first be converted to cart and then the user should follow the checkout process. So, we removed this mutation.